Coding/fw/lib/WebServer/ESP_Server.cpp

```diff
@@ -0,0 +1,251 @@
+
+    if (!dnsServer.start(WebConfig::DNS_PORT, "*", m_serverIP))
+    {
+        success = false;
+    }
+    else if (!SPIFFS.begin())
+    {
+        success = false;
+    }
+
+    Serial.println(m_serverIP);
+
+    server.begin();
+    return success;
+}
+
+/**************************************************************************************************/
+
+/**
+*   Disconnects and disables the WebServer
+*/
+bool ESPServer::end()
+{
+    server.end();
+    SPIFFS.end();
+    dnsServer.stop();
+    return WiFi.disconnect(true, true);
+}
+
+/**************************************************************************************************/
+/**
+*   Calls next request on DNS Server
+*   return true
+*/
+bool ESPServer::handleNextRequest()
+{
+    if (WiFi.status() != WL_CONNECTED)
+    {
+        if(!connectWiFi())
+        {
+            Board::errorLED.write(HIGH);
+            while(true)
+            {}
+        }
+    }
+    else
+    {
+        dnsServer.processNextRequest();
+        restartRequested = CaptivePortal::isRestartRequested();
+    }
+
+    return restartRequested;
+}
+
+AsyncWebServer &ESPServer::getInstance()
+{
+    return server;
+}
+
+/* PROTECTED METHODES *****************************************************************************/
+
+/* PRIVATE METHODES *******************************************************************************/
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+
+/* INTERNAL FUNCTIONS *****************************************************************************/
+/**************************************************************************************************/
+static bool setAPMode()
+{
+
+    bool apMode = false;
+
+    uint8_t currentBtnState = LOW;
+    uint8_t previousBtnState = LOW;
+    uint32_t lastDebounceTime = 0;
+
+    const uint8_t DEBOUNCE_DELAY = 50;
+    const uint32_t SETUP_TIME = 2000;
+    const uint32_t START_TIME = millis();
+
+    while ((millis() - START_TIME) < SETUP_TIME)
+    {
+        currentBtnState = Board::wifiModeSelect.read();
+
+        if (currentBtnState != previousBtnState)
+        {
+            lastDebounceTime = millis();
+        }
+
+        if ((millis() - lastDebounceTime) > DEBOUNCE_DELAY)
+        {
+            apMode = currentBtnState;
+        }
+
+        previousBtnState = currentBtnState;
+    }
+
+    return apMode;
+}
+
+/**************************************************************************************************/
+bool initPages(bool apModeRequested)
+{
+    bool success = true;
+
+    if (apModeRequested)
+    {
+        CaptivePortal::init(server);
+    }
+    else
+    {
+        Pages::init(server);
+    }
+
+    return success;
+}
+
+bool connectWiFi()
+{
+    bool success = true;
+
+    unsigned long startAttempTime = millis();
+
+    while (WiFi.status() != WL_CONNECTED && (millis() - startAttempTime) < WebConfig::WIFI_TIMEOUT_MS)
```

Comment:
add brackets also around the && operater parts. Don't rely on operater precedence rules.